### PR TITLE
Rrao/august integration/deviceregistrationrepo tests

### PR DESCRIPTION
**Pull request by rraomsft**

### CODE REVIEW

File: DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceRegistryRepositoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+            var deviceId = devices.First().DeviceProperties.DeviceID;
+
+            this._mockDocumentDBClient
+                .Setup(x => x.QueryAsync())
+                .ReturnsAsync(devices.AsQueryable());
+
+            var device = await this._deviceRegistryRepository.GetDeviceAsync(deviceId);
+
+            Assert.NotNull(device);
+            Assert.Same(devices[0], device);
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceRegistryRepositoryTests.cs]: we should use Assert.Same in some previous tests as well as it does reference matching and not content matching

File: DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceRegistryRepositoryTests.cs [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+            await Assert.ThrowsAsync<DeviceAlreadyRegisteredException>(() => this._deviceRegistryRepository.AddDeviceAsync(devices.First()));
+        }
+
+        [Fact]
+        public async Task AddDeviceAsync_should_add_device_to_db()
+        {
+            var device = this._fixture.Create<DeviceModel>();
+            await this._deviceRegistryRepository.AddDeviceAsync(device);
+            this._mockDocumentDBClient.Verify(x => x.SaveAsync(device));
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceRegistryRepositoryTests.cs]: you haven't mocked the SaveAsync call ?

File: DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceRegistryRepositoryTests.cs [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        [Fact]
+        public async Task RemoveDeviceAsync_should_throw_device_not_found_when_device_does_not_exist()
+        {
+            this._mockDocumentDBClient
+                .Setup(x => x.QueryAsync())
+                .ReturnsAsync(Enumerable.Empty<DeviceModel>().AsQueryable());
+
+            await Assert.ThrowsAsync<DeviceNotRegisteredException>(() => this._deviceRegistryRepository.RemoveDeviceAsync("foobar"));
+
+            this._mockDocumentDBClient.Verify();
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceRegistryRepositoryTests.cs]: mark function call you want to verify as Verifiable()

File: DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceRegistryRepositoryTests.cs (hunk 2 above)
Comment (rraomsft) [inline DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceRegistryRepositoryTests.cs]: I don't have to. Mocks do auto call counting so mocking is not explicitly concept.

File: DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceRegistryRepositoryTests.cs (hunk 3 above)
Comment (rraomsft) [inline DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceRegistryRepositoryTests.cs]: Verify auto verifies any previously .Setup calls. So no explicit Verify(x = x.QueryAsync()) is required.

Comment (rraomsft) [inline DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceRegistryRepositoryTests.cs]: Fixed. Using VerifyAll which should be the correct verification call.

**Pull request merged**